pkg/pipeline/encode/encode_prom.go

```diff
@@ -211,19 +319,28 @@ func NewEncodeProm() (Encoder, error) {
 				log.Errorf("error during prometheus.Register: %v", err)
 				return nil, err
 			}
-			histograms[metricInfo.Name] = histInfo{metricInfo.ValueKey, labels, hist}
+			histograms[metricInfo.Name] = genericMetricInfo{
+				input:      metricInfo.ValueKey,
+				tags:       labels,
+				metricType: api.PromEncodeOperationName("Histogram"),
+				promHist:   hist,
+			}
 		case "default":
 			log.Errorf("invalid metric type = %v, skipping", metricInfo.Type)
 		}
 	}
 
 	w := &encodeProm{
-		port:       fmt.Sprintf(":%v",portNum),
+		port:       fmt.Sprintf(":%v", portNum),
 		prefix:     promPrefix,
 		counters:   counters,
 		gauges:     gauges,
 		histograms: histograms,
+		expiryTime: expiryTime,
+		mList:      list.New(),
+		mCache:     make(metricCache),
 	}
 	go startPrometheusInterface(w)
+	go w.cleanupExpiredEntriesLoop()
```

Comment:
There are nicer ways to implement a background thread with a timer ... something like https://levelup.gitconnected.com/timer-and-ticker-using-golang-933a6d00a832

Using a channel and timer instead of forever loop with sleep

Reply:
I changed it to use a ticker instead of sleep.

Reply:
OK ... there are more "fancy" things to add here but this is ok for now :-) the other things are more like clearing the thread when the code exits ... but this is not critical for us.